frontend/src/pages/ReceiptCollabPage.tsx

```
@@ -35,9 +42,42 @@ const ReceiptCollabPage = () => {
+    ) {
+      setNotFoundStartTime(Date.now());
+      setRetryCount(0);
+    } else if (data?.[0]) {
+      // Receipt found, reset tracking
+      setNotFoundStartTime(null);
+      setRetryCount(0);
+    }
+  }, [details.type, data, notFoundStartTime]);
+
+  // Handle retry logic - wait between retries before giving up
+  useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)